File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
def bag_count(n):
    bag_count = 0
    
    while n >= 0:
        if n % 5 == 0: 
            bag_count += n // 5
            return bag_count
        n -= 3  
        bag_count += 1
    
    return -1  

number = int(input())
print(bag_count(number))

File: 백준/Silver/2839. 설탕 배달/설탕 배달.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
const input = fs.readFileSync(filePath).toString().trim();

function solution(input) {
  const n = Number(input);

  let result = -1;

  for (let i = Math.floor(n / 5); i >= 0; i--) {
    let left = input - i * 5;

    if (left % 3 === 0) {
      let j = left / 3;
      result = i + j;
      break;
    }
  }

  return result;
}

console.log(solution(input));
